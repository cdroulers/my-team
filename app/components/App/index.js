/**
 *
 * App.js
 *
 * This component is the skeleton around the actual pages, and should only
 * contain code that should be seen on all pages. (e.g. navigation bar)
 *
 */

import React from "react";
import { Switch, Route } from "react-router-dom";

import HomePage from "components/HomePage/Loadable";
import NotFoundPage from "components/NotFoundPage/Loadable";

import GlobalStyle from "../../global-styles";

function Test() {
  return <div>hai, this is a test page. lkjsdfjklsdf</div>;
}

export default function App() {
  return (
    <div>
      TEST
      <Switch>
        <Route exact path="/" component={HomePage} />
        <Route exact path="/test" component={Test} />
        <Route component={NotFoundPage} />
      </Switch>
      <GlobalStyle />
    </div>
  );
}
