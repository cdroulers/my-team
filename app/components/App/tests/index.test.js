import React from "react";
import ShallowRenderer from "react-test-renderer/shallow";
import { Provider as UnstatedProvider } from "unstated";
import { MemoryRouter } from "react-router-dom";

import App from "../index";

const renderer = new ShallowRenderer();

describe("<App />", () => {
  it("should render and match the snapshot", () => {
    renderer.render(
      <MemoryRouter>
        <UnstatedProvider>
          <App />
        </UnstatedProvider>
      </MemoryRouter>,
    );
    const renderedOutput = renderer.getRenderOutput();
    expect(renderedOutput).toMatchSnapshot();
  });
});
