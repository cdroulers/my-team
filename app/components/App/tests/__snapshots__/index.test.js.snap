// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should render and match the snapshot 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "canGo": [Function],
      "createHref": [Function],
      "entries": Array [
        Object {
          "hash": "",
          "key": "ok24ir",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
      ],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "index": 0,
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "key": "ok24ir",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <Provider>
    <App />
  </Provider>
</Router>
`;
